text/0000-blight-lang.md

```diff
@@ -0,0 +1,159 @@
+interface with JavaScript and we plan to do that in a future proposal.
+
+In the best timeline, once we get the compiler up and running, we are able to
+create a runtime pretty fast and kick off a hackathon to get an ecosystem
+running. If we get some key members of the JS community together, we may be able
+to get most of the core necessary modules done pretty fast.
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+This proposal is mainly to set up the motivation, goals, and non-goals of the
+language. The motivation has been thoroughly explained in the previous section,
+but let's go over goals and non-goals.
+
+Goals:
+
+- ESM-like modules
+- JS-like async
+- Rust-like syntax
+- Macros
+- Transpiles to JS
+- Fast to compile and produces relatively fast JS
+- Standard library to interact with raw JS APIs safely
+- Zig-like "safe-but-slow" debug build and "fast-but-has-UB" production build
+
+Non-Goals
+
+- Support for any JS libraries outside of wrappers (would just be a worse
+  Typescript)
+- Compile to standalone executable (we'd just be writing Rust again)
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+The main consideration for this RFC on the technical side is how will this all
+be implemented. My thinking at the current moment is that we write a
+transpiler + type checker in Rust just for performance sake. I don't think there
+is a universe where self-hosting makes sense unless we somehow get a compiler
+backend.
+
+We could choose other languages as well, but I believe that any other choice is
+the wrong move:
+
+- Zig: Cool technical features, but too unstable to use. Bun suffers actively
+  with breaking changes.
+- C++: It's C++. If I wanted to shoot myself in the foot I wouldn't be in
+  support of Blight.
+- C: Too low level, too much UB, hard to port.
+- Javascript/Typescript/Python: Too slow, too big of a binary
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+Maybe starting from scratch with no ecosystem is the wrong play. I could be
+entirely wrong about this. Deno regretted starting from scratch and there is a
+strong likelihood we will be in the same boat. If we can weather the storm, the
+waters look much clearer on the other side.
+
+# Rationale and alternatives
+
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+There are some alternatives that are tempting for sure. We could just... not do
+this project and accept the status quo. We could make something else, maybe this
+problem is too hard. I think "it's easier to do the other thing" is the wrong
+way of making long-term decisions. I truly believe starting from scratch is the
+right move here.
+
+I have considered (and gotten decently far) at implementing alternative runtimes
+for Javascript. I thought that maybe it was an API-level issue, but it's simply
+not. At a fundamental language level, Javascript is broken. It is unsafe. It is
+a footgun.
```

Comment:
It would be great if you provided a small bullet-list of examples to support this claim

Reply:
A bullet-list would be interesting. Off of the top of my head, `==` and `typeof null` come up immediately in my head. It would be valuable to write a list, I agree.